jwst/cube_build/blot_cube_build.py

```diff
@@ -131,98 +131,144 @@ def blot_images(self):
                 #mask out the side channel we aren not working on 
                 pixel_mask = np.full(model.shape,False,dtype=bool)
                 pixel_mask[:,xstart:xend] = True
+                ra_det,dec_det,lam_det = model.meta.wcs(xdet,ydet)
 
             elif self.instrument =='NIRSPEC':
-                blot.meta.filename = filename[:indx] + '_blot.fits' #set output name
-                ydet,xdet=np.mgrid[:2048,:2048]
-
-# determine the ra and dec values for each x,y pixel on the detector.
-
+                blot.meta.filename = filename[:indx] + '_blot.fits' 
+                ra_det=np.zeros((2048,2048))
+                dec_det=np.zeros((2048,2048))
+                lam_det=np.zeros((2048,2048))
+                flag_det = np.zeros((2048,2048))
+                
+                # for NIRSPEC each file has 30 slices - 
+                # wcs information access seperately for each slice
+                start_slice = 0
+                end_slice = 29
+                nslices = end_slice - start_slice + 1
+                regions = list(range(start_slice, end_slice + 1))
+#                log.info('Looping over 30 slices on NIRSPEC detector, this takes a little while')
+                for ii in regions:
+                    slice_wcs = nirspec.nrs_wcs_set_input(model, ii)
+                    x,y = wcstools.grid_from_bounding_box(slice_wcs.bounding_box)
+                    ra, dec, lam = slice_wcs(x, y)
+
+                    valid1 = np.isfinite(ra)
+                    valid2 = np.isfinite(dec)
+                    valid3 = np.isfinite(lam)
+                    value = valid1 & valid2 & valid3
+                    good_data =  np.where( value == True)
+                    
+                    ra = ra[good_data]
+                    dec = dec[good_data]
+                    lam = lam[good_data]
+                    x = x[good_data]
+                    y = y[good_data]
+                    
+                    xind = x.astype(np.int)
+                    yind = y.astype(np.int)
+                    xind = np.ndarray.flatten(xind)
+                    yind = np.ndarray.flatten(yind)
```

Comment:
These are indices but they are supposed to be pixels and depending which python function you use rounding is slightly different. np.int I think just removes the decimal part. So there's a function which can be used to keep going from floating value pixels to indices.

```
from jwst.tarnsforms.models import _toindex
xind = _toindex(x)
```